keymap/sublime.js

```diff
@@ -480,6 +480,112 @@
     modifyWordOrSelection(cm, function(str) { return str.toLowerCase(); });
   };
 
+  // Breaks up a given string into a list of "words" regardless of its original
+  // case.
+  //
+  // This function isn't very clever, but it does the right thing for all
+  // of the following standard cases:
+  //
+  // * snake_case
+  // * SCREAMING_SNAKE_CASE
+  // * lowerCamelCase
+  // * UpperCamelCase
+  // * Space seperated words
+  // * dot.case
+  // * dash-case
+  //
+  // Outside of these cases, your milageage will vary, however this function
+  // should perform a reasonably sensible split.
+  //
+  // Some examples:
+  //
+  // this_is_an_example -> [this, is, an, example]
+  //
+  // camelCaseIsSupported -> [camel, Case, Is, Supported]
+  //
+  // alsoSupportsAcronymsLikeABCNicely
+  //    -> [also, Supports, Acronyms, Like, ABC, Nicely]
+  //
+  // supports-mixed.delimiters for  better_or    worse
+  //    -> [supports, mixed, delimiters, for, better, or, worse]
+  //
+  // repeated--delimiters--are--ignored -> [repeated, delimiters, are, ignored]
+  function splitStringToRedoCase(str) {
+    return str
+      // This part splits up camelCased portions of the string,
+      // seperating them with whitespace.
+      .replace(/([a-z])([A-Z])/g, '$1 $2')
+      .replace(/([A-Z])([a-z])/g, ' $1$2')
+      // Trim out any leading or trailing whitespace
+      // The replaces above sometimes produce some.
+      .trim()
```

Comment:
We're still targeting ES3 runtimes, so I'm afraid you'll have to do the trimming manually.